src/bitcoin-cli.cpp

```diff
@@ -275,9 +273,6 @@ class GetinfoRequestHandler: public BaseRequestHandler
             }
             result.pushKV("paytxfee", batch[ID_WALLETINFO]["result"]["paytxfee"]);
         }
-        if (!batch[ID_BALANCES]["result"].isNull()) {
```

Comment:
Any reason to drop this (and break existing scripts)? Especially when `-rpcwallet` is set - and in this case all balances aren't even necessary.

Reply:
Thanks for reviewing @promag. Hm, I don't know if API stability with -getinfo is an issue if it's intended for human use. With just the default wallet loaded the PR currently displays:
```
{
  "version": 209900,
  "blocks": 15599,
  "headers": 15599,
  "verificationprogress": 1,
  "timeoffset": 0,
  "connections": 0,
  "proxy": "",
  "difficulty": 4.656542373906925e-10,
  "chain": "regtest",
  "keypoolsize": 1000,
  "paytxfee": 0.00000000,
  "relayfee": 0.00001000,
  "warnings": "This is a pre-release test build - use at your own risk - do not use for mining or merchant applications",
  "balances": {
    "": 0.00000000
  }
}
```
ISTM it's better to display the balances consistently, whether one wallet or several?

Reply:
Maybe I'm used to it, but I like having the balances set off in braces... it's easier to see them.

Reply:
I mean that you could leave "balance" untouched and just add `balances`. Later we could drop it. Only a suggestion, but
 1. if `-rpcwallet` is set why show all balances? Sounds conceptually wrong. Note that `getwalletinfo` uses that (or are you planning to call `getwalletinfo` for all wallets?
 2. this is not an API but it's a command people can use in scripts so why break it for no good reason?

Reply:
Yes, `getwalletinfo` adds a "walletname" field when `-rpcwallet` is set, but -getinfo does not. (FWIW it's odd that some field names use underscores and others don't... I'm guessing the snakecased ones are newer). It's late but I'll think about this.

Reply:
Note that if multiple wallets are loaded and if `-rpcwallet` is not set then `getwalletinfo` fails, which results in not displaying `keypoolsize` and `paytxfee`.

So I think if `-rpcwallet` is set, `balances` shouldn't be displayed.

Reply:
>     2. this is not an API but it's a command people can use in scripts so why break it for no good reason?

Most of the proposed ideas in #17314 could break scripts. Not breaking scripts would mean we would have to add things without removing or replacing them. This would make -getinfo unnecessarily long, but keeping -getinfo as short as possible is also a goal I've seen stated by long-term contributors.

Reply:
> Note that if multiple wallets are loaded and if `-rpcwallet` is not set then `getwalletinfo` fails, which results in not displaying `keypoolsize` and `paytxfee`.
> 
> So I think if `-rpcwallet` is set (or just one wallet is loaded), `balances` shouldn't be displayed.

Ok, I will try this out and likely update the PR with this.

Reply:
Done